pkg/landscaper/installations/imports/constructor.go

```diff
@@ -67,10 +80,20 @@ func (c *Constructor) Construct(ctx context.Context, inst *installations.Install
 				if def.Required != nil && !*def.Required {
 					continue // don't throw an error if the import is not required
 				}
-				return installations.NewImportNotFoundErrorf(nil, "no import for %s exists", def.Name)
+				return nil, installations.NewImportNotFoundErrorf(nil, "no import for %s exists", def.Name)
 			}
 			if err := c.JSONSchemaValidator().ValidateGoStruct(def.Schema.RawMessage, imports[def.Name]); err != nil {
-				return installations.NewErrorf(installations.SchemaValidationFailed, err, "%s: imported datatype does not have the expected schema", defPath.String())
+				return imports, installations.NewErrorf(installations.SchemaValidationFailed, err, "%s: imported datatype does not have the expected schema", defPath.String())
+			}
+			if def.ConditionalImports != nil && len(def.ConditionalImports) > 0 {
```

Comment:
in go slices that are nil do also have a length of `0`
```suggestion
			if len(def.ConditionalImports) > 0 {
```